docs/ref/models/expressions.txt

```diff
@@ -588,22 +588,38 @@ expressions::
 
 Next, we write the method responsible for generating the SQL::
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, **kwargs):
         sql_expressions, sql_params = [], []
         for expression in self.expressions:
             sql, params = compiler.compile(expression)
             sql_expressions.append(sql)
             sql_params.extend(params)
         self.extra['expressions'] = ','.join(sql_expressions)
```

Comment:
I would add the `data = kwargs.copy()` way of implementing `as_sql`, very similar to the actual implementation of `Func.as_sql` unless you think that'd confuse the implementation? This method here is mutating `self.extra` which we're trying to advocate against.